executor/src/subscriber.rs

```diff
@@ -1,95 +1,103 @@
 // Copyright (c) 2022, Mysten Labs, Inc.
```

Comment:
Theoretically I could eliminate the `Subscriber` as well and embed the functionality in the `executor::Core` . However, the separation with the `Subscriber` offers us buffering capabilities, ie if the `Core` for some reason stuck we can still continue downloading the certificates' batches and output in the `tx_executor` channel, effectively saving time. Of course if `tx_executor` fills up we'll stall as well, but that's desired.

Reply:
we may not need a separate thread for the subscriber but I would keep the logic in the core as simple as possible (so Subscriber may remain but perhaps not spawn its own tokio task?)

Reply:
hhm what do you mean no spawn it's own tokio task? I mean, if we don't want to make subscriber block we _need to_ run our loop in a task. Unless you mean something different?

Reply:
We can either keep the subscriber as it is (in this PR) or make it a collection of functions that are called by the Executor Core